src/Components/UpcomingTable/UpcomingRow.tsx

```diff
@@ -0,0 +1,130 @@
+import "@patternfly/react-core/dist/styles/base.css";
+
+import React, { useState } from "react";
+import {
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  ExpandableRowContent
+} from "@patternfly/react-table";
+
+export const TableRow: React.FunctionComponent = ({
+  repo,
+  rowIndex,
+  columnNames,
+  
+}) => {
+  const [isRepoExpanded, setIsRepoExpanded] = useState(false);
+  // Some arbitrary examples of how you could customize the child row based on your needs
+  let childIsFullWidth = false;
+  let childHasNoPadding = false;
+  let detail1Colspan = 1;
+  let detail2Colspan = 1;
+  let detail3Colspan = 1;
+  if (repo.details) {
+    const { detail1, detail2, detail3, detailFormat } = repo.details;
+    const numColumns = 5;
+    childIsFullWidth = [1, 3].includes(detailFormat);
+    childHasNoPadding = [2, 3].includes(detailFormat);
+    if (detail1 && !detail2 && !detail3) {
+      detail1Colspan = !childIsFullWidth ? numColumns : numColumns + 1; // Account for toggle column
+    } else if (detail1 && detail2 && !detail3) {
+      detail1Colspan = 2;
+      detail2Colspan = !childIsFullWidth ? 3 : 4;
+    } else if (detail1 && detail2 && detail3) {
+      detail1Colspan = 2;
+      detail2Colspan = 2;
+      detail3Colspan = !childIsFullWidth ? 1 : 2;
+    }
+  }
+  return (
+    <Tbody key={`${repo.name}-${repo.type}-${repo.release}-${repo.date}`} 
+    isExpanded={isRepoExpanded}>
+      <Tr>
+        <Td
+          expand={{
+            rowIndex: rowIndex,
+            isExpanded: isRepoExpanded,
+            onToggle: () => setIsRepoExpanded(!isRepoExpanded),
+            expandId: "composable-expandable-example"
+          }}
+        />
+        <Td dataLabel={columnNames.name} modifier="truncate">
+            {repo.name}
+        </Td>
+        <Td dataLabel={columnNames.type} modifier="truncate">
+            {repo.type}
+        </Td>
+        <Td dataLabel={columnNames.release} modifier="truncate">
+            {repo.release}
+        </Td>
+        <Td dataLabel={columnNames.date} modifier="truncate">
+            {repo.date}
+        </Td>
+      </Tr>
+      {repo.details ? (
+        <Tr isExpanded={isRepoExpanded}>
+            {!childIsFullWidth ? <Td /> : null}
+            {repo.details.summary ? (
```

Comment:
We can remove these checks. This is saying `{if repo.details.summary is not undefined then (render the cell) or render nothing}`

If we know `repo.details.summary` will always return in the api, then we don't need this check. Otherwise, I think there is a better way to handle it.